storm-core/src/jvm/org/apache/storm/serialization/KryoTupleDeserializer.java

```diff
@@ -50,7 +54,8 @@ public Tuple deserialize(byte[] ser) {
             List<Object> values = _kryo.deserializeFrom(_kryoInput);
             return new TupleImpl(_context, values, taskId, streamName, id);
         } catch(IOException e) {
-            throw new RuntimeException(e);
+            LOG.error("Failed to deserialize tuple.", e);
```

Comment:
In our case this happens because storm's internal message exchange protocol is vulnerable to foreign  attack. If deserialization fails, you should warn the user and go on with processing.